divide_and_conquer/strassen_matrix_multiplication.py

```diff
@@ -112,17 +112,19 @@ def strassen(matrix1: list, matrix2: list) -> list:
     [[139, 163], [121, 134], [100, 121]]
     """
     if matrix_dimensions(matrix1)[1] != matrix_dimensions(matrix2)[0]:
-        raise Exception(
-            "Unable to multiply these matrices, please check the dimensions. \n"
-            f"Matrix A:{matrix1} \nMatrix B:{matrix2}"
+        msg = (
+            "Unable to multiply these matrices, please check the dimensions.\n"
+            f"Matrix A: {matrix1}\n"
+            f"Matrix B: {matrix2}"
         )
+        raise Exception(msg)
```

Comment:
What is the point of this change? I preferred it without this change

Reply:
One of the new ruff rules introduced in #8767 disallows f-strings in error messages

Reply:
> One of the new ruff rules introduced in #8767 disallows f-strings in error messages

I see
https://beta.ruff.rs/docs/rules/f-string-in-exception/